intro-OpenStack/ch_ConfigureInstances.tex

```diff
@@ -0,0 +1,154 @@
+
+If you have generated a key pair with an external tool, you can import
+it into \gls{OpenStack}. The key pair can be used for multiple
+instances that belong to a project. For more information, see the
+section \emph{Import a key pair}.
+
+\strong{Note:} A key pair belongs to an individual user, not to a
+project. To share a key pair across multiple users, each user needs to
+import that key pair.
+
+\strong{Note:} This chapter only explains the required configuration
+in order to make your instance accessible via SSH.  For instructions
+on how to connect to a running instance, once it has been configured
+correctly, see page \pageref{connect-to-your-instance-using-ssh} of
+chapter \ref{cha:launch-manage-inst}.
+
+\subsubsection{Add a key pair}\label{add-a-key-pair}
+\begin{enumerate}
+\item Open the Compute tab.
+\item Click the Key Pairs tab, which shows the key pairs that are
+  available for this project.
+\item Click Create Key Pair.
+\item In the Create Key Pair dialog box, enter a name for your key
+  pair, and click Create Key Pair.
+\item Respond to the prompt to download the key pair.
+\item Save the \textbf{*.pem} file locally.
+\item To change its permissions so that only you can read and write to
+  the file, run the following command:
+
+  \begin{prompt}
+      %\shellcmd{chmod 0600 yourPrivateKey.pem}
+  \end{prompt}
+
+  \strong{Note:} If you are using the \gls{OpenStack Dashboard} from a
+  Windows computer, use PuTTYgen to load the \textbf{*.pem} file and
+  convert and save it as \textbf{*.ppk}.  For more information see the
+  \href{https://winscp.net/eng/docs/ui_puttygen}{\emph{WinSCP web page
+      for PuTTYgen}}, and chapter 2 of
+  \href{https://www.vscentrum.be/support/tut-book/vsc-tutorials}{the
+    VSC HPC tutorial}.
+
+\item To make the key pair known to SSH, run the \textbf{ssh-add}
+  command.
+
+  \begin{prompt}
+    %\shellcmd{ssh-add yourPrivateKey.pem}
+  \end{prompt}
+\end{enumerate}
+
+\subsubsection{Import a key pair}\label{import-a-key-pair}
+\begin{enumerate}
+\item Open the Compute tab.
+\item Click the Key Pairs tab, which shows the key pairs that are
+  available for this project.
+\item Click Import Key Pair.
+\item In the Import Key Pair dialog box, enter the name of your key
+  pair, copy the public key into the Public Key box, and then click
+  Import Key Pair.
+\end{enumerate}
+
+The Compute database registers the public key of the key pair.
+
+The \gls{OpenStack Dashboard} lists the key pair on the Key Pairs tab.
+
+\subsubsection{Allocate a floating IP address to an instance}\label{allocate-a-floating-ip-address-to-an-instance}
+When an instance is created in \gls{OpenStack}, it is automatically
+assigned a fixed IP address in the network to which the instance is
+assigned. This IP address is permanently associated with the instance
+until the instance is terminated.
+
+However, in addition to the fixed IP address, a floating IP address
```

Comment:
@tdanckaert @alvarosimon associating floating ip address to an instance should be considered advanced/non-default usage. the default behaviour should be to setup port forwarding as it is more flexible and safer

Reply:
I think that we should keep both methods in the same section:
* option a) using port forwarding (recommended) 
and
* option b) assign the floating directly to a VM (advance/testing usage)

Reply:
I've moved all info on port forwarding and attaching to the "configure access and security" chapter, focusing on the port forwarding part.

By the way: the drop-down next to the floating ip's still has the "Release Floating IP" option.  I believe this will remove the FIP from the pool for the project, something the regular users cannot undo... can we remove this option from the interface?  The current docs also contain a note about "Release Floating IP", taken from the original docs.  Should we remove that note?  Or turn it into warning not to do that?

Reply:
@tdanckaert we should include a warning about that, do not release the floating ip. If they do that they have to ask us to assign again the IP to their project.
We have to remove the "Release Floting Ip" checkbox but this should be done in the python code (it is not supported directly from the dashboard conf as we did to remove the extra tabs)

Reply:
Ok, that's done.

I've also removed some information about allocation of new IP's to the project's floating IP pool, because our users can't do this anyway.  The Dashboard still has the options to do this, but I think it might be more complicated to remove those.